internal/impl/devbox.go

```diff
@@ -252,7 +248,7 @@ func (d *Devbox) RunScript(ctx context.Context, cmdName string, cmdArgs []string
 		return err
 	}
 
-	if err := d.writeScriptsToFiles(); err != nil {
+	if err := filegen.WriteScriptsToFiles(d); err != nil {
```

Comment:
I guess the difference here would be, if another package imports our code directly and initializes a `(d *devbox)` object. these functions are no longer exposed, since `filegen` is strictly internal?

Reply:
Hmm I'm less concerned about the scenario you mention.

My motivation was that I wanted to pull logic out of `impl` and `plansdk` into a single package (and internalize parts of it so it is easier to reason about). Currently, we have logic for generating code in `.devbox/gen` that is spread in different `.go` files in both `impl` and `plansdk`. (I'll add this to the summary)